src/internal/cluster/zarf.go

```diff
@@ -99,50 +99,42 @@ func (c *Cluster) StripZarfLabelsAndSecretsFromNamespaces() {
 }
 
 // PackageSecretNeedsWait checks if a package component has a running webhook that needs to be waited on.
-func (c *Cluster) PackageSecretNeedsWait(deployedPackage types.DeployedPackage, component types.ZarfComponent, skipWebhooks bool) (needsWait bool, waitSeconds int, hookName string, err error) {
+func (c *Cluster) PackageSecretNeedsWait(deployedPackage *types.DeployedPackage, component types.ZarfComponent, skipWebhooks bool) (needsWait bool, waitSeconds int, hookName string) {
 
-	// Skip checking webhook status when '--skip-webhooks' flag is provided
-	if skipWebhooks {
-		return false, 0, "", nil
-	}
-
-	// Skip checking webhook status on YOLO packages
-	if deployedPackage.Data.Metadata.YOLO {
-		return false, 0, "", nil
+	// Skip checking webhook status when '--skip-webhooks' flag is provided and for YOLO packages
+	if skipWebhooks || deployedPackage.Data.Metadata.YOLO {
```

Comment:
```suggestion
	if skipWebhooks || deployedPackage == nil || deployedPackage.Data.Metadata.YOLO {
```
Might be good to have a nil check as well in case this is used in library contexts